muon/_prot/preproc.py

```diff
@@ -202,21 +205,39 @@ def clr(adata: AnnData, inplace: bool = True) -> Union[None, AnnData]:
         data: AnnData object with protein expression counts.
         inplace: Whether to update adata.X inplace.
     """
+
+    if not inplace:
+        adata = adata.copy()
+
     sparse = False
     if issparse(adata.X):
         sparse = True
     # Geometric mean of ADT counts
     x = adata.X
-    g_mean = np.exp(np.log1p(x).sum(axis=0) / x.shape[0])
-    # Centered log ratio
-    clr = np.log1p(x / g_mean)
 
-    if sparse:
-        clr = csr_matrix(clr)
+    if flavor == "seurat":
+        if issparse(x):
+            x.data /= np.repeat(
+                np.exp(np.log1p(x).sum(axis=1).A / x.shape[1]),
+                x.getnnz(axis=1),
+            )
+            np.log1p(x.data, out=x.data)
```

Comment:
This inplace solution requires `x` to be in CSR format. The type could be checked beforehand and converted to CSR if needed. The inplace operation will no longer work then, i.e. `adata.X` would have to be set explicitly again.

Reply:
I may be wrong, but this should also work for CSC matrices, right? In any case. a check of the matrix type should definitely be done, with either a warning and an automatic conversion or an informative error message.

Reply:
The `np.repeat()` part will produce the same result. The normalization sizes are calculated and ordered by row. However, for a CSC matrix `x.data` is sorted by column. Hope this makes it clear. The following code snippet should run as is and will throw an `AssertionError`:

```python
import numpy as np
from scipy.sparse import csc_matrix, csr_matrix

np.random.seed(0)
csc = csc_matrix(np.random.randint(low=0, high=10, size=(10, 4)))

np.random.seed(0)
csr = csr_matrix(np.random.randint(low=0, high=10, size=(10, 4)))

a = csc.data / np.repeat(csc.sum(axis=1).A / csc.shape[1], csc.getnnz(axis=1))
b = csr.data / np.repeat(csr.sum(axis=1).A / csr.shape[1], csr.getnnz(axis=1))

np.testing.assert_equal(a, b)
```

```pytb
AssertionError: 
Arrays are not equal

Mismatched elements: 29 / 36 (80.6%)
Max absolute difference: 1.6
Max relative difference: 6.
 x: array([1.818182, 2.545455, 0.727273, 1.333333, 1.166667, 0.833333,
       0.666667, 1.052632, 1.684211, 0.631579, 1.894737, 0.695652,
       1.391304, 1.217391, 1.565217, 0.521739, 0.173913, 1.217391,...
 y: array([1.818182, 1.090909, 1.090909, 1.166667, 1.5     , 0.5     ,
       0.833333, 0.421053, 0.842105, 1.473684, 1.263158, 1.391304,
       1.391304, 0.173913, 1.043478, 1.217391, 1.217391, 1.391304,...
```